BackHandler Fix

kadirmertgul opened the pull request:
I made the necessary changes to fix crashes in my React Native project caused by the back button on Android devices and the closing of Awesome Alerts using BackHandler.

foxriver76 commented:
Any possibility to get this fix in @rishabhbhatia? without it the lib won't work for up-to-date RN projects.

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -23,6 +23,7 @@ const { OS } = Platform;
 export default class AwesomeAlert extends Component {
   constructor(props) {
     super(props);
+    this.backHandlerSubscription = null; // ✅ Değişkeni baştan tanımla
```
```suggestion
    this.backHandlerSubscription = null;
```

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -103,9 +107,9 @@ export default class AwesomeAlert extends Component {
     } = data;
 
     return (
-      <TouchableOpacity  style={[styles.button, { backgroundColor }, buttonStyle]} testID={testID} onPress={onPress}>
```
this seems unnecessary, can we fix the linting here back to original

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -166,31 +170,31 @@ export default class AwesomeAlert extends Component {
     };
 
     return (
-      <View style={[styles.container, alertContainerStyle]}>
```
this seems unnecessary, can we fix the linting here back to original

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -201,22 +205,22 @@ export default class AwesomeAlert extends Component {
-        </Modal>
-      ) : this._renderAlert()
-    : null;
+        wrapInModal ? (
```
same feedback for linting, keep the original

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -228,8 +232,9 @@ export default class AwesomeAlert extends Component {
 
   componentWillUnmount() {
-    HwBackHandler.removeEventListener(HW_BACK_EVENT, this._handleHwBackEvent);
+    this.backHandlerSubscription?.remove(); // ✅ Eğer varsa kaldır
```
```suggestion
    this.backHandlerSubscription?.remove();
```

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -228,8 +232,9 @@ export default class AwesomeAlert extends Component {
-    HwBackHandler.removeEventListener(HW_BACK_EVENT, this._handleHwBackEvent);
+    this.backHandlerSubscription?.remove(); // ✅ Eğer varsa kaldır
   }
+
```
seems unnecessary, can we keep the original

rishabhbhatia commented:
I see a `First-time contributor` badge on @kadirmertgul, so i'd love for you to have this contribution commit. I'll wait a bit to see if you have the time and resources to make the changes needed here. Overall the change makes sense to me!

@foxriver76 for sure, happy to merge these and release a new version once the necessary changes are made on this PR.

kadirmertgul commented:
Hey @rishabhbhatia, thanks for the feedback!

✅ I’ve made all the requested changes:

Removed all the // comments for clarity.

Reverted the unnecessary lint changes back to the original formatting.

Set showSelf to initialize from props.show as suggested.

Let me know if there’s anything else I should adjust. Thanks again for your time and for considering the PR!

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -23,18 +23,22 @@ const { OS } = Platform;
-    this.state = {
-      showSelf: false,
-    };
+   this.state = {
```
this needs to be reverted back to how it was before

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -103,9 +107,9 @@ export default class AwesomeAlert extends Component {
 
     return (
-      <TouchableOpacity  style={[styles.button, { backgroundColor }, buttonStyle]} testID={testID} onPress={onPress}>
+        <TouchableOpacity  style={[styles.button, { backgroundColor }, buttonStyle]} testID={testID} onPress={onPress}>
```
there's still an extra space here that makes this change show up in the diff of the PR. If you remove that extra spacing, this would not show up here. Thats the state we'd want to be in.
```suggestion
      <TouchableOpacity  style={[styles.button, { backgroundColor }, buttonStyle]} testID={testID} onPress={onPress}>
```

rishabhbhatia commented on src/AwesomeAlert.js:
```diff
@@ -166,31 +170,31 @@ export default class AwesomeAlert extends Component {
-          </View>
-        </Animated.View>
-      </View>
+        <View style={[styles.container, alertContainerStyle]}>
```
same as above, the green square you're seeing is the extra unwanted spacing that we want to remove. This change should not show up in the PR diff.

rishabhbhatia reviewed (changes requested):
thank you for making those changes. Same feedback as before, we'd want to remove all the extra code linting that's bloating the diff. One additional comment on an extra change made to `showSelf`.

If i had to guess, it might be your IDE auto-linting the file with your local settings once you finish making the changes. Look for the option to save without formatting. With the current change you would see the test fail too.

my expectation from this PR would be to consist a change for initialize the backhandler, subscribe to it when component mounts and remove the subscription on unmount.
